### DISCUSSION TITLE
Customize the session data?

### DISCUSSION THREAD

**Discussion by m1212e** [zitadel/zitadel#6115] [Q&A] [answered]
Hi,

I'm new to IAM in general and currently dig into Zitadel. I would like to store some per user data which I'd like to use in my APIs. I would put this data into a session, but I'm not sure if this is possible with Zitadel? How would one set, manage and store user data which is domain specific?

Thanks!

**Comment by m1212e** [comment]
Do I use https://zitadel.com/docs/apis/actions/complement-token to add my own claims? Or is this a bad practice?

**Comment by hifabienne** [comment, marked as answer]
Hei @m1212e 
If you have user specific data you want to store. We recommend using the user metadata.
Its basically a key value store, to add your data to a user.
You can also request it in the token by adding the metadata scope into the auth request.

https://zitadel.com/docs/guides/manage/customize/user-metadata
https://zitadel.com/docs/apis/openidoauth/scopes#reserved-scopes
